src/App.js

```diff
@@ -1,19 +1,126 @@
 import React from 'react';
 import './App.css';
+import UserList from './Components/UserList/UserList';
 
 import users from './api/users';
+import todos from './api/todos';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+const list = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId).name,
+}));
+
+class App extends React.Component {
+  state = {
+    todos: [...list],
+    title: '',
+    name: '',
+    todoError: false,
+    userError: false,
+  }
+
+  handleChangeTitle = (event) => {
+    this.setState({
+      title: event.target.value,
+    });
+
+    if (event.target.value !== '') {
+      this.setState({
+        todoError: false,
+      });
+    }
+  }
+
+  handleChangeName = (event) => {
+    this.setState({
+      name: event.target.value,
+    });
+
+    if (event.target.value !== '') {
+      this.setState({
+        userError: false,
+      });
+    }
+  }
+
+  handleFormSubmit = (event) => {
+    if (this.state.title === '') {
+      this.setState({ todoError: true });
+    }
+
+    if (this.state.name === '') {
+      this.setState({ userError: true });
+    }
+
+    if (this.state.name !== '' && this.state.title !== '') {
+      this.setState(state => ({
+        todos: [
+          ...state.todos,
+          {
+            user: state.name,
+            title: state.title,
+            completed: false,
+            userId: users.find(user => user.name === state.name).id,
+            id: state.todos.length + 1,
+          },
+        ],
+      }));
+
+      this.setState({
+        name: '',
+        title: '',
+        todoError: false,
+        userError: false,
+      });
+    }
+
+    event.preventDefault();
```

Comment:
This should be at the very start of the method.
If you'll have any async actions at the start(e.g. server requests), there may be some unexpected consequences.